### PR TITLE
[tests-only] refactor steps with password in them

### PR DESCRIPTION
**Pull request by swoichha**
## Description
This PR refactors all the steps with password in them

## Related Issue
- fixes https://github.com/owncloud/client/issues/8652

### CODE REVIEW

File: test/gui/shared/steps/steps.py [hunk 1]
```diff
@@ -87,9 +87,18 @@ def step(context, displayname, host):
...
+    userFirstname = userId[0]
+    if userFirstname in defaultUsers.keys():
+        return defaultUsers[userFirstname]['password']
+
+
+@Given('user "|any|" has set up a client with default settings')
+def step(context, username):
+    password = getPasswordForUser(username)
     setUpClient(context, username, confFilePath)
+    snooze(10)
```
Comment (dpakach) [inline test/gui/shared/steps/steps.py:101]: why is this required?

File: test/gui/shared/steps/steps.py [hunk 2]
```diff
@@ -87,9 +87,18 @@ def step(context, displayname, host):
     )
 
 
-@Given('user "|any|" has set up a client with default settings and password "|any|"')
-def step(context, username, password):
+def getPasswordForUser(username):
+    userId = username.split()
+    userFirstname = userId[0]
```
Comment (dpakach) [inline test/gui/shared/steps/steps.py:92]: I dont think this is required
this is equivalent to `userFirstname = username`

**Review by dpakach — CHANGES_REQUESTED** [review]

File: test/gui/shared/steps/steps.py (hunk 1 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py]: fixed

File: test/gui/shared/steps/steps.py (hunk 2 above)
Comment (swoichha) [reply test/gui/shared/steps/steps.py:92]: It is for username which has both first and last name like `Alice Hansen` used here https://github.com/owncloud/client/blob/808284d75bb23a85de3c8068051995dbbf8e18d2/test/gui/tst_addAccount/test.feature#L23

Comment (dpakach) [reply test/gui/shared/steps/steps.py:92]: That's the display name of the user. the username cannot have spaces I think it would be better to either use only username or only displayname in these kinds of steps.

Comment (swoichha) [reply test/gui/shared/steps/steps.py:92]: https://github.com/owncloud/client/pull/8656/files#diff-821cad5db62830778c291de069b5182ff9bcdd90951faa2cec9d223e8efbcc20R91 will split the display-name and will only be using the first-name which is also the username

**Review by individual-it — APPROVED** [review]

**Review by dpakach — APPROVED** [review]
